worlds/sc2/rules.py

```diff
@@ -29,6 +35,121 @@ def is_item_placement(self, state):
         # has_group with count = 0 is always true for item placement and always false for SC2 item filtering
         return state.has_group("Missions", self.player, 0)
 
+    # Needs to react on how many missions actually got generated
+    def total_mission_count(self):
+        return 1 if (self.world is None or not hasattr(self.world, 'custom_mission_order')) \
+            else self.world.custom_mission_order.get_mission_count()
+
+    # Unit classes in world, these are set properly into the world after the item culling is done
+    # Therefore need to get from the world
+    def world_has_barracks_unit(self):
+        if self.world is None:
+            return False
+        else:
+            return self.world.has_barracks_unit
```

Comment:
(minor, matter of taste) if you want to compress these functions down to one line, you can do `self.world and self.world.has_barracks_unit`

Reply:
changed to:
```python
    def world_has_barracks_unit(self):
        return self.world is not None and self.world.has_barracks_unit
```